crates/swc_ecma_transforms_compat/tests/__swc_snapshots__/tests/es2017_async_to_generator.rs/issue_8452.js

```diff
@@ -2,14 +2,8 @@ class Test0 {
 }
 class Test extends Test0 {
     constructor(){
-        var _this;
-        super(), _this = this, console.log(/*#__PURE__*/ function() {
-            var _ref = _async_to_generator(function*(e) {
-                yield _this.test();
-            });
-            return function(e) {
-                return _ref.apply(this, arguments);
-            };
-        }());
+        super(), console.log((e)=>/*#__PURE__*/ _async_to_generator(function*() {
```

Comment:
By injecting `this` at the call site instead of hoisting it, we avoid potential order issues of `this` and `super`.